File: VB/ReportParameterMVC/Controllers/HomeController.vb
Imports System
Imports System.Web.Mvc
Imports ReportParameterMVC.Reports

Namespace ReportParameterMVC.Controllers

    Public Class HomeController
        Inherits Controller

        Public Function Index(ByVal parameterOrderID As Integer?) As ActionResult
            ViewBag.Message = "Report Parameter Example"
            ' Add a report to a view data.
            Dim Report As XtraReport1 = New XtraReport1()
            ViewData("Report") = Report
            If parameterOrderID Is Nothing Then parameterOrderID = 10248 ' default parameter value
            Dim param As DevExpress.XtraReports.Parameters.Parameter = Report.Parameters("OrderID")
            If param IsNot Nothing Then param.Value = Convert.ToInt32(parameterOrderID)
            Return View()
        End Function

        Public Function ReportViewerPartial() As ActionResult
            ViewData("Report") = New XtraReport1()
            Return PartialView("ReportViewerPartial")
        End Function

        Public Function ExportReportViewer() As ActionResult
            Return DevExpress.Web.Mvc.ReportViewerExtension.ExportTo(New XtraReport1())
        End Function
    End Class
End Namespace

File: CS/ReportParameterMVC/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using ReportParameterMVC.Models;
using ReportParameterMVC.Reports;

namespace ReportParameterMVC.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index(int? parameterOrderID)
        {
            ViewBag.Message = "Report Parameter Example";
            // Add a report to a view data.
            XtraReport1 Report = new XtraReport1();
            ViewData["Report"] = Report;
            if (parameterOrderID == null)
                parameterOrderID = 10248; // default parameter value

            DevExpress.XtraReports.Parameters.Parameter param = Report.Parameters["OrderID"];
            if (param != null)
                param.Value = Convert.ToInt32(parameterOrderID);

            return View();
        }


        public ActionResult ReportViewerPartial()
        {
            ViewData["Report"] = new ReportParameterMVC.Reports.XtraReport1();
            return PartialView("ReportViewerPartial");
        }

        public ActionResult ExportReportViewer()
        {
            return DevExpress.Web.Mvc.ReportViewerExtension.ExportTo(new ReportParameterMVC.Reports.XtraReport1());
        }
    }
}
